test/testhelper/client/k8s/client.go

```diff
@@ -0,0 +1,284 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const (
+	ReadyStatus = "Ready"
+	TRUE        = "True"
+	FALSE       = "False"
+)
+
+type Client struct {
+	client *kubernetes.Clientset
+}
+
+type NamespacePod struct {
+	Namespace v1.Namespace
+	PodList   *v1.PodList
+}
+
+type EventPod struct {
+	Reason    string
+	Message   string
+	Count     int32
+	Type      string
+	Action    string
+	Namespace string
+}
+
+func NewK8sClient(sshClient *testhelper.SSHClient) (*Client, error) {
+	kubeconfigPath := filepath.Join("/root", ".kube", "config")
+
+	data := testhelper.GetFileDataRemote(sshClient, kubeconfigPath)
+	reg := regexp.MustCompile(`server: https://(.*):6443`)
+	data = reg.ReplaceAll(data, []byte("server: https://"+sshClient.RemoteHostIP.String()+":6443"))
+	config, err := clientcmd.RESTConfigFromKubeConfig(data)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to build kube config")
+	}
+
+	clientSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{
+		client: clientSet,
+	}, nil
+}
+
+func (c *Client) ListNodes() (*v1.NodeList, error) {
+	nodes, err := c.client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get cluster nodes")
+	}
+	return nodes, nil
+}
+
+func (c *Client) listNamespaces() (*v1.NamespaceList, error) {
+	namespaceList, err := c.client.CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get namespaces")
+	}
+	return namespaceList, nil
+}
+
+func (c *Client) ListNodesByLabel(label string) (*v1.NodeList, error) {
+	nodes, err := c.client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{LabelSelector: label})
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get cluster nodes")
+	}
+	return nodes, nil
+}
+
+func (c *Client) ListNodeIPByLabel(label string) ([]net.IP, error) {
+	var ips []net.IP
+	nodes, err := c.ListNodesByLabel(label)
+	if err != nil {
+		return nil, err
+	}
+	for _, node := range nodes.Items {
+		for _, v := range node.Status.Addresses {
+			if v.Type == v1.NodeInternalIP {
+				ips = append(ips, net.ParseIP(v.Address))
+			}
+		}
+	}
+	return ips, nil
+}
+
+func (c *Client) ListAllNamespacesPods() ([]*NamespacePod, error) {
+	namespaceList, err := c.listNamespaces()
+	if err != nil {
+		return nil, err
+	}
+	var namespacePodList []*NamespacePod
+	for _, ns := range namespaceList.Items {
+		pods, err := c.client.CoreV1().Pods(ns.Name).List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to get all namespace pods")
+		}
+		namespacePod := NamespacePod{
+			Namespace: ns,
+			PodList:   pods,
+		}
+		namespacePodList = append(namespacePodList, &namespacePod)
+	}
+
+	return namespacePodList, nil
+}
+
+func (c *Client) ListKubeSystemPodsStatus() (bool, error) {
```

Comment:
Actually, I don't think the function name fits what the function truly does.
The function name makes the callers feel it will return a list(KubeSystemPodsStatus), but it checks if all pods of "KubeSystem" are ready.